lib/src/test/java/space/taran/arklib/data/index/IndexAggregationTest.kt

```diff
@@ -0,0 +1,171 @@
+        every { rootIndex.updates } returns MutableSharedFlow<ResourceUpdates>()
+        every { rootIndex.allResources() } returns resources.associateBy({ it.id }, { it })
+        every { rootIndex.getResource(capture(resourceIdSlot)) } answers {
+            val resourceId = resourceIdSlot.captured
+            resources.firstOrNull { resource -> resource.id == resourceId }
+        }
+        every { rootIndex.getPath(capture(resourceIdSlot)) } answers {
+            val resourceId = resourceIdSlot.captured
+            resourcePaths[resourceId]
+        }
+        coEvery { rootIndex.updateAll() } answers {
+            updatedShards += 1
+        }
+        return rootIndex;
+    }
+
+    private fun createRandomResources(amount: Int): Set<Resource> {
+        return (1..amount).map {
+            val resourceId = createRandomResourceId()
+            Resource(resourceId, UUID.randomUUID().toString(), "txt", FileTime.from(Instant.now()))
+        }.toSet()
+    }
+
+    private fun createRandomResourceId(): ResourceId {
+        val crc32 = generateRandomLong(18)
+        val size = Random.nextLong(0, 16364)
+        return ResourceId(size, crc32)
+    }
+
+    private fun generateRandomLong(length: Int): Long {
+        require(length > 0) { "Length must be greater than 0" }
+        val stringBuilder = StringBuilder(length)
+        repeat(length) {
+            val digit = Random.nextInt(1, 10)
+            stringBuilder.append(digit.toStr())
+        }
+        return stringBuilder.toString().toLong()
+    }
+
+    private fun createResourceRandomPaths(resources: Set<Resource>): Map<ResourceId, Path> {
+        return resources.map { resource ->
+            val path = Path(UUID.randomUUID().toString())
+            resource.id to path
+        }.associateBy({ it.first }, { it.second })
+    }
+
+    @Test
+    fun testAllResources() = runBlocking {
+        val expected = shards.map { it.allResources() }.toList().reduce { acc, map -> acc + map }
+        val allResources = indexAggregation.allResources()
+        assertEquals(9, allResources.size) // 3 resources per shard, so 3 shards * 3 resources
+        assertEquals(expected, allResources)
+    }
+
+    @Test
+    fun testAllResourcesEmpty() = runBlocking {
+        val indexAggregation = IndexAggregation(setOf())
+        val allResources = indexAggregation.allResources()
+        assertEquals(0, allResources.size)
+        assertEquals(mapOf<ResourceId, Resource>(), allResources)
+    }
+
+    @Test
+    fun testGetResource() = runBlocking {
+        val shard = shards.random()
+        val resource = shard.allResources().values.random()
+        val result = indexAggregation.getResource(resource.id)
+        assertEquals(resource, result)
+    }
+
+    @Test
+    fun testGetResourceNotFound() = runBlocking {
+        val indexAggregation = IndexAggregation(setOf())
+        val result = indexAggregation.getResource(ResourceId(0, 123456789))
```

Comment:
Ids with zero size are forbidden actually. Let's create dedicated test case for this somewhere (not really relevant to aggregation).

About this test case, let's create a constant for an "arbitrary id which is not in the index".